src/particles/distribution/SpinvMF.H

```diff
@@ -0,0 +1,146 @@
+            amrex::ParticleReal & AMREX_RESTRICT sx,
+            amrex::ParticleReal & AMREX_RESTRICT sy,
+            amrex::ParticleReal & AMREX_RESTRICT sz,
+            amrex::RandomEngine const & engine
+        ) const
+        {
+            using namespace amrex::literals;
+            using amrex::Math::powi;
+            using ablastr::constant::math::pi;
+
+            // Normalize the mean direction vector just in case:
+            amrex::ParticleReal mu_norm = std::sqrt(powi<2>(m_muX)+powi<2>(m_muY)+powi<2>(m_muZ));
+            amrex::ParticleReal mux = m_muX/mu_norm;
+            amrex::ParticleReal muy = m_muY/mu_norm;
+            amrex::ParticleReal muz = m_muZ/mu_norm;
+
+            // Evaluate constants independent of particle:
+            amrex::ParticleReal const muperp = std::sqrt(powi<2>(mux)+powi<2>(muy));
+            amrex::ParticleReal const a = (muperp==0) ? 0_prt : mux/muperp;
+            amrex::ParticleReal const b = (muperp==0) ? 1_prt : muy/muperp;
+
+            // Sample two iid random variables:
+            amrex::ParticleReal x1 = amrex::Random(engine);
+            amrex::ParticleReal x2 = amrex::Random(engine);
+
+            // Basic transformation of the random variables:
+            amrex::ParticleReal c = std::cos(2_prt*pi*x1);
+            amrex::ParticleReal s = std::sin(2_prt*pi*x1);
+            amrex::ParticleReal t;
+            // The form of the function below is obtained from eq (30) of:
+            // D. Frisch and U. D. Hanebeck, "Deterministic Von Mises-Fisher Sampling on the Sphere Using
+            // Fibonacci Lattices", IEEE, 2023, DOI: 10.1109/SDF-MFI59545.2023.10361396
+            if(m_kappa > 0) {
+               t = 1_prt + std::log1p(x2 * std::expm1(-2_prt * m_kappa)) / m_kappa;
+            } else {
+               t = 1_prt - 2_prt * x2;
+            }
+
+            // Evaluation of the spin components
+            amrex::ParticleReal u = std::sqrt(1_prt-powi<2>(t));
+            sx = u * (b*c + a*muz*s) + t*mux;
+            sy = u * (-a*c + b*muz*s) + t*muy;
+            sz = u * (-muperp*s) + t*muz;
+
+        }
+
+
+        /** This function evaluatates the inverse Langevin function, in order to return
+         *  the value of concentration (kappa) required to produce a given polarization magnitude.
+         *  It is used to prepare for spin sampling using the vMF distribution.
+         *
+         *  Here, we use an approximate expression due to R. Petrosyan.  See:
+         *  R. Petrosyan, Rheologica Acta 56, 21-26, 2016; doi:10.1007/s00397-016-0977-9
+         *  R. Jedynak, Mathematics and Mechanics of Solids 24, 1-25, 2018; doi:10.1177/1081286518811395
+         *
+         * @param pmag polarization magnitude
+         * @returns required value of concentration kappa
+         */
+        AMREX_GPU_HOST AMREX_FORCE_INLINE
+        static amrex::ParticleReal
+        inverse_Langevin (amrex::ParticleReal const & AMREX_RESTRICT pmag)
```

Comment:
Just checking I see this correctly (see above):
`inverse_Langevin` handles values between 0 and 1 well, in particular:
- including 0 (result is zero)
- excluding 1 (result would be NaN from div by zero)

?